src/App.tsx

```diff
@@ -1,26 +1,198 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useRef, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { USER_ID, deleteTodo, getTodos, postTodo } from './api/todos';
+import classNames from 'classnames';
+import { Todo } from './types/Todo';
+import { TodoItem } from './components/Todo/Todo';
+import { Filter } from './types/Filter';
+import { TodoField } from './components/Todo/TodoField/TodoField';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [filter, setFilter] = useState(Filter.All);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [shouldFocus, setShouldFocus] = useState(false);
+  const timer = useRef(0);
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(() => setErrorMessage('Unable to load todos'));
+  }, []);
+
+  timer.current = window.setTimeout(() => setErrorMessage(''), 3000);
+
+  const hideError = () => {
+    clearTimeout(timer.current);
+    setErrorMessage('');
+  };
+
+  const addPost = (title: string): Promise<void> => {
+    setErrorMessage('');
+
+    return postTodo(title)
+      .then((newTodo: Todo) => {
+        setTodos(currentTodos => [...currentTodos, newTodo]);
+      })
+      .catch(error => {
+        setErrorMessage('Unable to add a todo');
+        throw error;
+      });
+  };
+
+  const onDelete = (id: number) => {
+    return deleteTodo(id)
+      .then(() => {
+        setTodos(currentTodos => currentTodos.filter(todo => todo.id !== id));
+        setShouldFocus(true);
+      })
+      .catch(error => {
+        setErrorMessage('Unable to delete a todo');
+        throw error;
+      });
+  };
+
+  const onCompleteDelete = useMemo(() => {
+    return () => {
+      const ids: number[] = [];
+
+      todos.filter(todo => todo.completed).forEach(todo => ids.push(todo.id));
+
+      for (let i = 0; i < ids.length; i++) {
+        onDelete(ids[i]);
+      }
```

Comment:
You can make this logic first forEach iteration
```suggestion
      todos.filter(todo => todo.completed).forEach(todo => onDelete(todo.id));

```